src/tpx/Sub.cpp

```diff
@@ -370,14 +430,16 @@ void Substance::Set(PropertyPair::type XY, double x0, double y0)
                x0, y0, TolAbsS, TolAbsH, TolRel, TolRel);
         break;
     case PropertyPair::PX:
-        temp = Tsat(x0);
+        temp = Tmin();
         if (y0 > 1.0 || y0 < 0.0) {
             throw CanteraError("Substance::Set",
                                "Invalid vapor fraction, {}", y0);
-        } else if (temp >= Tcrit()) {
+        } else if (x0 < Ps() || x0 > Pcrit()) {
```

Comment:
Ah ... the temperature is set to `Tmin()` on line 433 right above, which in theory would ensure a correct execution, but I figure I forgot to call `set_T` on that.

Regarding pre-calculation: the constructor isn't really used at the moment, so there's not a good 'hook'. I'm also afraid that it will make #605 rear its head again (currently I don't have spare time to address).